File: Data Structures & Algorithms/string-encode-and-decode/submission-2.py
class Solution:

    def encode(self, strs: List[str]) -> str:
        if not strs:    return []
        res = []
        for s in strs:
            length = len(s)
            res.append(f"{length}#{s}")
        return ''.join(res)
        
    def decode(self, s: str) -> List[str]:
        res = []
        i = 0
        while i < len(s):
            j = i
            while s[j] != "#":
                j += 1
            length = int(s[i : j])
            res.append(s[j + 1 : j + length + 1])
            i = j + length + 1
        return res

File: Data Structures & Algorithms/reverse-bits/submission-2.cpp
class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        uint32_t res = 0;
        for (int i = 31; i >= 0; i--) {
            if (n & (1 << (31 - i))) {
                res += (1 << i);
            }
        }

        return res;
    }
};
